src/map.c

```diff
@@ -1,754 +1,628 @@
 /*
  * rv32emu is freely redistributable under the MIT License. See the file
  * "LICENSE" for information on usage and redistribution of this file.
+ * This map implementation is heavily modified from rb.h of jemalloc.
  */
 
-#include <assert.h>
-#include <stdint.h>
-#include <stdlib.h>
-#include <string.h>
-
-#include "map.h"
-
-struct map_internal {
-    struct map_node *head;
-
-    /* Properties */
-    size_t key_size, element_size, size;
-
-    map_iter_t it_end, it_most, it_least;
-
-    int (*comparator)(const void *, const void *);
-};
-
-typedef enum { RB_RED = 0, RB_BLACK } map_color_t;
-
 /*
- * Get parent of node
- * @node: pointer to the rb node
- *
- * Return: rb parent node of @node
+ * This map implementation is heavily modified based on the rb.h header file of
```

Comment:
Proofread the comment as following:
> This map implementation has undergone extensive modifications, heavily relying on the rb.h header file from jemalloc. See https://github.com/jemalloc/jemalloc/blob/dev/include/jemalloc/internal/rb.h .
The original rb.h file served as the foundation and source of inspiration for adapting and tailoring it specifically for this map implementation. Therefore, credit and sincere thanks are extended to jemalloc for their invaluable work.